decentraland/redux/reducers.js

```diff
@@ -85,51 +85,69 @@ function assetsForSale(state = { list: [], loadingInProgress: true }, action) {
 }
 
 function myAssets(state = { list: [] }, action) {
-  const {
-    type,
-    myAsset,
-    myAssets,
-    myAssetAndActionIds,
-    myAssetAndStatus,
-    itemOrList,
-  } = action;
+  const { type } = action;
 
   switch (type) {
     case PREPEND_TO_MY_ASSETS_LIST:
-      return { ...state, list: [myAsset, ...state.list] };
-    case APPEND_TO_MY_ASSETS_LIST: {
-      const toAppend = toListIfNot(itemOrList);
-      return { ...state, list: [...state.list, ...toAppend] };
-    }
-    case REMOVE_FROM_MY_ASSETS_LIST: {
-      const { list: myAssets } = state;
-      const toRemove = itemOrList;
-      const list = removeFromList(myAssets, toRemove);
-      return { ...state, list };
-    }
-    case SET_MY_ASSETS_LIST: {
-      const list = myAssets === null ? [] : myAssets;
-      return { ...state, list };
-    }
-    case SET_ACTION_ID_FOR_MY_ASSET: {
-      const { myAssetId: toUpdateId, actionId } = myAssetAndActionIds;
-      const { list: myAssets } = state;
-      const entriesToUpdate = { actionId };
-      const list = updateListItem(myAssets, toUpdateId, entriesToUpdate);
-      return { ...state, list };
-    }
-    case UPDATE_MY_ASSET_STATUS: {
-      const { myAssetId: toUpdateId, status } = myAssetAndStatus;
-      const { list: myAssets } = state;
-      const entriesToUpdate = { status };
-      const list = updateListItem(myAssets, toUpdateId, entriesToUpdate);
-      return { ...state, list };
-    }
+      return prependToMyAssetsList(state, action);
+    case APPEND_TO_MY_ASSETS_LIST:
+      return appendToMyAssetsList(state, action);
+    case REMOVE_FROM_MY_ASSETS_LIST:
+      return removeFromMyAssetsList(state, action);
+    case SET_MY_ASSETS_LIST:
+      return setMyAssetsList(state, action);
+    case SET_ACTION_ID_FOR_MY_ASSET:
+      return setActionIdForMyAsset(state, action);
+    case UPDATE_MY_ASSET_STATUS:
+      return updateMyAssetStatus(state, action);
     default:
       return state;
   }
 }
 
+const prependToMyAssetsList = (state, action) => {
+  const { myAsset } = action;
```

Comment:
Is there a version of this where the action can just be the relevant piece of data for the reducer that will handle it? It seems redundant to also have to unpack a key/value that we already could have guessed based on the action name

Reply:
Hmm, I think I take that back. Because the name of the action is also part of the action object

Reply:
I've found this on the [redux docs](https://redux.js.org/recipes/structuring-reducers/refactoring-reducer-example#combining-reducers-by-slice):
```javascript
function createReducer(initialState, handlers) {
  return function reducer(state = initialState, action) {
    if (handlers.hasOwnProperty(action.type)) {
      return handlers[action.type](state, action)
    } else {
      return state
    }
  }
}

const todosReducer = createReducer([], {
  ADD_TODO: addTodo,
  TOGGLE_TODO: toggleTodo,
  EDIT_TODO: editTodo
})
```

Reply:
Ah yeah, exactly. Up to you whether that feels like a useful change to make or not.

Reply:
Done.

Reply:
Great, thanks. Resolving this.